internal/bytesconv/bytesconv.go

```diff
@@ -5,16 +5,17 @@
 package bytesconv
 
 import (
-	"reflect"
 	"unsafe"
 )
 
 // StringToBytes converts string to byte slice without a memory allocation.
-func StringToBytes(s string) (b []byte) {
-	sh := *(*reflect.StringHeader)(unsafe.Pointer(&s))
-	bh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
-	bh.Data, bh.Len, bh.Cap = sh.Data, sh.Len, sh.Len
-	return b
+func StringToBytes(s string) []byte {
```

Comment:
What is the principle behind this change?

Reply:
@panjf2000 based on https://github.com/gin-gonic/gin/issues/2553, we also need to change it.